File: reference/protocol/src/main/java/com/nagada/pulse/protocol/SyncResponse.java
package com.nagada.pulse.protocol;

import java.util.List;

/**
 * Sync response from server: acknowledgments + new server events.
 */
public class SyncResponse {
    public List<String> ackedClientEventIds;
    public List<ServerEvent> newServerEvents;

    public SyncResponse() {
    }

    public SyncResponse(List<String> ackedClientEventIds, List<ServerEvent> newServerEvents) {
        this.ackedClientEventIds = ackedClientEventIds;
        this.newServerEvents = newServerEvents;
    }
    //add getters 
    public List<String> getAckedClientEventIds() {
        return ackedClientEventIds;
    }
    public List<ServerEvent> getNewServerEvents() {
        return newServerEvents;
    }
}

File: client/dart/nagada_client/lib/core/models/sync_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'server_event.dart';

part 'sync_response.g.dart';

/// Data Transfer Object for a successful sync response from the server.
@JsonSerializable(explicitToJson: true, fieldRename: FieldRename.snake)
class SyncResponse {
  /// List of client event IDs that the server has successfully processed.
  final List<String> ackedClientEventIds;

  /// List of new events from the server that the client needs to apply.
  final List<ServerEvent> newServerEvents;

  SyncResponse({
    required this.ackedClientEventIds,
    required this.newServerEvents,
  });

  /// Creates a new `SyncResponse` from a JSON map.
  factory SyncResponse.fromJson(Map<String, dynamic> json) =>
      _$SyncResponseFromJson(json);

  /// Converts this `SyncResponse` to a JSON map.
  Map<String, dynamic> toJson() => _$SyncResponseToJson(this);
}